typing/ctype.ml

```diff
@@ -2736,7 +2747,7 @@ let constrain_type_jkind ~fixed env ty jkind =
        substitutions into the with-bounds while in [estimate_type_jkind]. These
        substitutions can be expensive if there are many with-bounds, and this
        case gets hit a lot (the v1 safety check always hits this case). *)
-    let ignore_mod_bounds = Jkind.mod_bounds_are_max jkind in
+    let ignore_mod_bounds = Jkind.mod_bounds_are_max env jkind in
     let ty's_jkind = estimate_type_jkind ~ignore_mod_bounds env ty in
     loop ~fuel ~expanded ty ty's_jkind jkind
```

Comment:
`Jkind.mod_bounds_are_max` expands the jkind, but we pass the unexpanded jkind on to `loop`. As a result, we do some redundant expansions (although I doubt this is very important).